gcloud-java-pubsub/src/main/java/com/google/gcloud/pubsub/spi/SubscriberApi.java

```diff
@@ -0,0 +1,663 @@
+            ListSubscriptionsRequest payload, Object token) {
+          return ListSubscriptionsRequest
+            .newBuilder(payload)
+            .setPageToken((String) token)
+            .build();
+        }
+        @Override
+        public Object extractNextToken(ListSubscriptionsResponse payload) {
+          return payload.getNextPageToken();
+        }
+        @Override
+        public Iterable<Subscription> extractResources(ListSubscriptionsResponse payload) {
+          return payload.getSubscriptionsList();
+        }
+      };
+
+  private static String ALL_SCOPES[] = {
+    "https://www.googleapis.com/auth/pubsub"
+  };
+
+  /**
+   * A PathTemplate representing the fully-qualified path to represent
+   * a project resource.
+   */
+  public static final PathTemplate PROJECT_PATH_TEMPLATE =
+      PathTemplate.create("/projects/{project}");
+
+  /**
+   * A PathTemplate representing the fully-qualified path to represent
+   * a subscription resource.
+   */
+  public static final PathTemplate SUBSCRIPTION_PATH_TEMPLATE =
+      PathTemplate.create("/projects/{project}/subscriptions/{subscription}");
+
+  // ========
+  // Members
+  // ========
+
+  private final ManagedChannel channel;
+  private final ServiceApiSettings settings;
+
+  // ===============
+  // Factory Methods
+  // ===============
+
+  /**
+   * Constructs an instance of SubscriberApi with default settings.
+   */
+  public static SubscriberApi create() throws IOException {
+    return create(new ServiceApiSettings());
+  }
+
+  /**
+   * Constructs an instance of SubscriberApi, using the given settings. The channels are created based
+   * on the settings passed in, or defaults for any settings that are not set.
+   */
+  public static SubscriberApi create(ServiceApiSettings settings) throws IOException {
+    return new SubscriberApi(settings);
+  }
+
+  private SubscriberApi(ServiceApiSettings settings) throws IOException {
+    ServiceApiSettings internalSettings = ApiUtils.settingsWithChannels(settings,
+        SERVICE_ADDRESS, DEFAULT_SERVICE_PORT, ALL_SCOPES);
+    this.settings = internalSettings;
+    this.channel = internalSettings.getChannel();
+  }
+
+  // ==============================
+  // Resource Name Helper Functions
+  // ==============================
+
+  /**
+   * Creates a string containing the fully-qualified path to represent
+   * a project resource.
+   */
+  public static final String createProjectPath(String project) {
+    return PROJECT_PATH_TEMPLATE.instantiate(
+        "project", project);
+  }
+
+  /**
+   * Creates a string containing the fully-qualified path to represent
+   * a subscription resource.
+   */
+  public static final String createSubscriptionPath(String project, String subscription) {
+    return SUBSCRIPTION_PATH_TEMPLATE.instantiate(
+        "project", project,"subscription", subscription);
+  }
+
+
+  // ========
+  // Getters
+  // ========
+
+  public Channel getChannel() {
```

Comment:
There was a reason for this. The rules of engagement indicate that you should be able fall down to direct gRPC usage easily, so exposing the channel is part of this. If we remove in it, we should at least add an issue to discuss whether it should be added back.